grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -317,9 +321,25 @@ public synchronized void setMessageCompression(boolean enabled) {
     public void messageRead(DeframedMessage message) {
         if (isGrpcWeb && message.type() >> 7 == 1) {
             // grpc-web trailers
-            final ByteBuf buf = message.buf();
-            // trailers never compressed
-            assert buf != null;
+            final ByteBuf messageBuf = message.buf();
+            final ByteBuf buf;
+            if (messageBuf != null) {
+                buf = messageBuf;
+            } else {
+                buf = ctx.alloc().compositeBuffer();
```

Comment:
Any reason to use a composite buffer?

Reply:
It is hard to know about the sensible initial capacity because I do not know how much it's expanded after decompression.
All other logic uses the composite buffer, so I followed it.
Can I revisit this later to optimize this?

Reply:
Added TODO. 😉

Reply:
SGTM